### PR TITLE
Updated rust template

### PR DESCRIPTION
**Pull request by rajarshimaitra**
This adds the rust solution template.

key changes:

- Suggested dependencies changed to `bitcoin` and `bitcoincore-rpc`.
- Fix stuffs at `balance.rs`.
- New template at `spend.rs`.

Please suggest further improves.

### CODE REVIEW

File: solution/rust/balance/src/lib.rs [hunk 1]
```diff
@@ -1,44 +1,40 @@
...
 }
-
 // final wallet state struct
 pub struct WalletState {
-    utxos: Vec<OutgoingTx>,
-    witness_programs: Vec<[u8; 22]>,
-    public_keys: Vec<[u8; 33]>,
-    private_keys: Vec<[u8; 32]>,
+    utxos: Vec<TxOut>,
+    witness_programs: WitnessProgram,
```
Comment (pinheadmz) [inline solution/rust/balance/src/lib.rs:28]: should this also be `vec` ?

File: solution/rust/balance/src/lib.rs [hunk 2]
```diff
@@ -1,44 +1,40 @@
...
-const EXTENDED_PRIVATE_KEY: &str = tprv8ZgxMBicQKsPfCxvMSGLjZegGFnZn9VZfVdsnEbuzTGdS9aZjvaYpyh7NsxsrAc8LsRQZ2EYaCfkvwNpas8cKUBbptDzadY7c3hUi8i33XJ
-const HARDENED_OFFSET: u32 = 2_u32.pow(31);
 
-struct ExtendedKey {
+// Import necessary bitcoin primitives
+extern crate bitcoin;
+use bitcoin::{
+    bip32::{Xpriv, Xpub},
+    script::PushBytesBuf,
+    Amount, PrivateKey, PublicKey, ScriptBuf, Transaction, TxOut, WitnessProgram,
```
Comment (pinheadmz) [inline solution/rust/balance/src/lib.rs:9]: For both challenges, Id like students to define the serialization for objects like script, tx, witness program etc. and also implement the BIP32 math and base58 algorithm themselves. Can we remove this entire dependency?

File: solution/rust/balance/src/lib.rs (hunk 1 above)
Comment (rajarshimaitra) [reply solution/rust/balance/src/lib.rs]: Missed that. Done.

File: solution/rust/balance/src/lib.rs (hunk 2 above)
Comment (rajarshimaitra) [reply solution/rust/balance/src/lib.rs]: Ah I see. Removed all the rust-bitcoin imports and used std byte arrays instead.

**Pull request merged**